### PR TITLE
fix: #174 remove dependencies of gdc in filters container

### PR DESCRIPTION
**Pull request by francisl**
Prepare Filters to be in ferlab-ui 

things left to commit.
- Query builder
- Translation

### CODE REVIEW

File: src/packages/cqdg-ui/core/containers/filters/FilterContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  searchValue: string;
+  searchEnabled: boolean;
+}
+
+interface IFilterContainerState {
+  searchInputVisible: boolean;
+  collapsed: boolean;
+}
+
+class FilterContainer
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/FilterContainer.tsx:28]: we should move this to a functionnal component

File: src/packages/cqdg-ui/core/containers/filters/FilterContainerHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                  ) : null
+                }
+      >
+      <div onClick={() => onClick(!collapsed)} role="button">
+
+        {!angleIconRight && (
+          <AngleIcon
+            style={{
+              transform: `rotate(${collapsed ? 270 : 0}deg)`,
+            }}
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/FilterContainerHeader.tsx]: move to css file

File: src/packages/cqdg-ui/core/containers/filters/Filters.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+
+import ExactMatchFacet from 'cqdg-ui/core/containers/filters/types/ExactMatchFacet';
+
+import SingleChoice from 'cqdg-ui/core/containers/filters/types/SingleChoice';
+import MultipleChoice from 'cqdg-ui/core/containers/filters/types/MultipleChoice';
+import RangeFilter from 'cqdg-ui/core/containers/filters/types/RangeFilter';
+import t from '@cqdg/locales/intl';
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/Filters.tsx:8]: maybe im blind but I dont see it used

File: src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import { getFilterValue, makeFilter } from '@cqdg/utils/filters';
+
+import CheckCircleOIcon from 'react-icons/lib/fa/check';
+import { Tooltip } from '@cqdg/components/Tooltip';
+import Link from '@cqdg/components/Links/Link';
+
+const ExactMatchFacet = compose(
+  withState('inputValue', 'setInputValue', ''),
+  pure,
+)(
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js:22]: we should switch this to a functionnal component

File: src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                query={
+                        inputValue && {
+                          filters: makeFilter([
+                            {
+                              field: `${doctype}.${fieldNoDoctype}`,
+                              value: [inputValue],
+                            },
+                          ]),
+                        }
+                      }
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js:95]: I think the data should be all ready for the UI, there should be no manipulation to it in the UI

File: src/packages/cqdg-ui/core/containers/filters/types/MultipleChoice.js [hunk 6]
```diff
@@ -280,14 +278,13 @@ const enhance = compose(
       'searchValue',
     ],
     ({
-      buckets, filter, isMatchingSearchValue, searchValue = '',
+      buckets, filter, searchValue = '',
```
Comment (kjacqueschu) [inline src/packages/cqdg-ui/core/containers/filters/types/MultipleChoice.js:281]: We should move out of recompose in favor of hooks

**Review by kjacqueschu — APPROVED** [review]

File: src/packages/cqdg-ui/core/containers/filters/FilterContainer.tsx (hunk 1 above)
Comment (francisl) [reply src/packages/cqdg-ui/core/containers/filters/FilterContainer.tsx:28]: doesn't work in current implementation with useState

File: src/packages/cqdg-ui/core/containers/filters/Filters.tsx (hunk 3 above)
Comment (francisl) [reply src/packages/cqdg-ui/core/containers/filters/Filters.tsx:8]: https://github.com/Ferlab-Ste-Justine/cqdg-portal-ui/pull/94/files/16ccf262295b774b5d753a03763d2a647a87c177#diff-0353fbb4dd43054d2c6fd07c0e58bbf3f3c332ef1b7bbb8bd76e2852a4f534d6R84

File: src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js (hunk 5 above)
Comment (francisl) [reply src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js:95]: fixing that in the next PR

File: src/packages/cqdg-ui/core/containers/filters/types/MultipleChoice.js (hunk 6 above)
Comment (francisl) [reply src/packages/cqdg-ui/core/containers/filters/types/MultipleChoice.js:281]: hooks make the build fail. We cannot use theme until we move to the new repo of ferlab-ui

File: src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js (hunk 4 above)
Comment (francisl) [reply src/packages/cqdg-ui/core/containers/filters/types/ExactMatchFacet.js:22]: build failed with functional component with useState, we need to wait for the new repo

**Pull request merged**